### PR TITLE
update: add info to enable/disable Kafka SASL mechanisms

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
## Describe your changes
- PR updates the documentation to include instructions on how to enable and disable Kafka SASL mechanisms in Aiven for Apache Kafka®.
- Updated relevant pages with new information on Kafka SASL mechanisms.
- Enhanced documentation pages to use a tabbed approach for Console, CLI, and API methods.

Ref: [HH-4195](https://aiven.atlassian.net/browse/HH-4195)

## Checklist

- [x] The first paragraph of the page is on one line.
- [x] The other lines have a line break at 90 characters.
- [x] I checked the output.
- [x] I applied the [style guide](../styleguide.md).
- [x] My links start with `/docs/`.

[HH-4195]: https://aiven.atlassian.net/browse/HH-4195?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/products/kafka/howto/kafka-sasl-auth.md [hunk 1]
```diff
@@ -1,28 +1,184 @@
...
+
+   - `project_name`: Name of your Aiven project.
+   - `service_name`: Name of your Aiven for Apache Kafka service.
+   - `API_TOKEN`: API token for [authentication](/docs/platform/howto/create_authentication_token).
+   - `kafka_authentication_methods.sasl`: Set to `true` to enable SASL authentication.
+
+</TabItem>
+</Tabs>
+
+## Configure SASL mechanisms
```
Comment (juha-aiven) [inline docs/products/kafka/howto/kafka-sasl-auth.md:86]: Is it clear here that `kafka_sasl_mechanisms.plain` and the other two are by default enabled, i.e. if *not* configured, they're enabled. Or to put it the other way around, only if explicitly disabled they are disabled.

Comment (harshini-rangaswamy) [reply docs/products/kafka/howto/kafka-sasl-auth.md:86]: I updated the content to clarify this point further.

**Review by MichaelTansiniAiven — APPROVED** [review]
Looks good to me.

File: docs/products/kafka/concepts/auth-types.md [hunk 2]
```diff
@@ -2,66 +2,63 @@
...
 
-**PLAIN** relies on a combination of username and password to log in
-over a TLS connection, meaning that your traffic is encrypted. To ensure
-proper security mechanism Aiven does not support using SASL/PLAIN
-without TLS, because in this case anyone will be able to read your
-credentials when you send them.
 
-### SASL/SCRAM
+- **SCRAM-SHA-512**: Similar to SCRAM-SHA-256 but uses SHA-512 hashing for added
+  security. SCRAM works by creating a random `salt`, which is then used to create an
```
Comment (wojcik-dorota) [inline docs/products/kafka/concepts/auth-types.md:43]: nit:
```suggestion
  security. SCRAM works by creating a random `salt`, which is used to create an
```

File: docs/products/kafka/concepts/auth-types.md [hunk 3]
```diff
@@ -2,66 +2,63 @@
...
+**Transport Layer Security (TLS)**, also known as Secure Sockets Layer (SSL),
+is a standard for securing internet traffic. This method relies on a certificate
+provided by a Certificate Authority (for example,
+[letsencrypt.org](https://letsencrypt.org)). With this certificate and the right
+technical setup, you can use your domain to encrypt traffic to your service.
 
-By default Aiven enables TLS encryption for all **Aiven for Apache
-Kafka** services and helps with the application, renewal, and
-configuration of certificates.
+By default Aiven enables TLS encryption for all **Aiven for Apache Kafka** services
```
Comment (wojcik-dorota) [inline docs/products/kafka/concepts/auth-types.md:15]: nit:
```suggestion
By default, Aiven enables TLS encryption for all **Aiven for Apache Kafka** services
```

File: docs/products/kafka/howto/enable-oidc.md [hunk 4]
```diff
@@ -1,14 +1,27 @@
...
 token-based authentication and integrate with identity providers.
 Setting the JSON Web Key Set (JWKS) JWKS endpoint via the Aiven console
 activates the OIDC mechanism for Kafka, which triggers a rolling restart
 of the Kafka brokers. This restart does not cause service downtime.
 
+:::note
+To use the `OAUTHBEARER` mechanism, you must enable `kafka_authentication_methods.sasl`.
+Additionally, at least one of the SASL mechanisms (PLAIN, SCRAM-SHA-256, or SCRAM-SHA-512)
+must be enabled. For instructions on enabling SASL authentication, see
+[Enabled and configure SASL authentication with Aiven for Apache Kafka®](/docs/products/kafka/howto/kafka-sasl-auth)
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/enable-oidc.md:22]: Typo and some minor proposal how to make it shorter:
```suggestion
must be enabled. See
[Enable and configure SASL authentication with Aiven for Apache Kafka®](/docs/products/kafka/howto/kafka-sasl-auth).
```

File: docs/products/kafka/howto/enable-oidc.md [hunk 5]
```diff
@@ -18,108 +31,111 @@ official documentation for specific configuration guidelines.
...
-    As a result, the brokers may temporarily operate with different
-    configurations. To minimize any operational disruptions, plan to
-    implement these changes during a maintenance window or at a time
-    that ensures a minimal impact on your operations.
-    :::
-
-1.  Select **Save configurations** to save your changes
-
-## Enable OAuth2/OIDC via Aiven CLI
+- [Aiven for Apache Kafka®](/docs/products/kafka/get-started) service running.
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/enable-oidc.md:34]:
```suggestion
- [Aiven for Apache Kafka®](/docs/products/kafka/get-started) service running
```

File: docs/products/kafka/howto/enable-oidc.md [hunk 6]
```diff
@@ -18,108 +31,111 @@ official documentation for specific configuration guidelines.
...
+- **Access to an OIDC provider**: Options include Auth0, Okta, Google
+  Identity Platform, Azure, or any other OIDC compliant provider.
+- Required configuration details from your OIDC provider:
+  - **JWKS Endpoint URL**: URL to retrieve the JSON Web Key Set
+    (JWKS).
+  - **Subject Claim Name**: Typically `sub`. However, this can vary depending
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/enable-oidc.md:40]: nit:
```suggestion
  - **Subject Claim Name**: Typically `sub` but can vary depending
```

File: docs/products/kafka/howto/enable-oidc.md [hunk 7]
```diff
@@ -18,108 +31,111 @@ official documentation for specific configuration guidelines.
...
+  - **Audience identifiers**: Validates the JWT's intended
+    recipients. For multiple audiences, make a note of all.
+
+## Enable OAuth2/OIDC
+
+<Tabs groupId="config-methods">
+<TabItem value="console" label="Aiven Console" default>
+
+1. In the [Aiven Console](https://console.aiven.io/), select your
+   project and choose your Aiven for Apache Kafka® service.
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/enable-oidc.md:53]: nit:
```suggestion
   project and Aiven for Apache Kafka® service.
```

File: docs/products/kafka/howto/enable-oidc.md [hunk 8]
```diff
@@ -18,108 +31,111 @@ official documentation for specific configuration guidelines.
...
-
-For detailed explanations on the OIDC parameters, refer to the
-[Enable OAuth2/OIDC via Aiven Console](#console-authentication) section above.
+1. Get the name of your Aiven for Apache Kafka service:
+
+   ```bash
+   avn service list
+   ```
+
+   Note of the `SERVICE_NAME` corresponding to your Aiven for Apache Kafka service.
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/enable-oidc.md:74]:
```suggestion
   Note the `SERVICE_NAME` corresponding to your Aiven for Apache Kafka service.
```

File: docs/products/kafka/howto/kafka-sasl-auth.md [hunk 9]
```diff
@@ -1,28 +1,185 @@
...
+1. Access the [Aiven Console](https://console.aiven.io) and select your
+   Aiven for Apache Kafka service.
+1. Click <ConsoleLabel name="Service settings"/>.
+1. Scroll to **Advanced configuration** and click **Configure**.
+1. In the **Advanced configuration** window, set `kafka_authentication_methods.sasl` to
+   **Enabled**.
+1. Click **Save configurations**.
 
-![Choose between SASL and certificate connection details](/images/content/products/kafka/sasl-connect.png)
+The **Connection information** in the **Overview** page now allows connections via SASL
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/kafka-sasl-auth.md:25]:
```suggestion
The **Connection information** in the <ConsoleLabel name="overview"/> page now allows connections via SASL
```

File: docs/products/kafka/howto/kafka-sasl-auth.md [hunk 10]
```diff
@@ -1,28 +1,185 @@
...
+Enable SASL authentication for your Aiven for Apache Kafka service using
+[Aiven CLI](/docs/tools/cli):
+
+1. Get the name of the Aiven for Apache Kafka service:
+
+   ```bash
+   avn service list
+   ```
+
+   Note of the `SERVICE_NAME` corresponding to your Aiven for Apache Kafka service.
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/kafka-sasl-auth.md:45]:
```suggestion
   Note the `SERVICE_NAME` corresponding to your Aiven for Apache Kafka service.
```

File: docs/products/kafka/howto/kafka-sasl-auth.md [hunk 11]
```diff
@@ -1,28 +1,185 @@
...
+             }
+          }
+       }'
+   ```
+
+   Parameters:
+
+   - `project_name`: Name of your Aiven project.
+   - `service_name`: Name of your Aiven for Apache Kafka service.
+   - `API_TOKEN`: API token for [authentication](/docs/platform/howto/create_authentication_token).
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/kafka-sasl-auth.md:81]:
```suggestion
   - `project_name`: Name of your Aiven project
   - `service_name`: Name of your Aiven for Apache Kafka service
   - `API_TOKEN`: API token for [authentication](/docs/platform/howto/create_authentication_token)
```

File: docs/products/kafka/howto/kafka-sasl-auth.md [hunk 12]
```diff
@@ -1,28 +1,185 @@
...
+
+1. Enable SASL authentication:
+
+   ```bash
+   avn service update SERVICE_NAME -c kafka_authentication_methods.sasl=true
+   ```
+
+   Parameters:
+
+   - `SERVICE_NAME`: Name of your Aiven for Apache Kafka service.
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/kafka-sasl-auth.md:55]:
```suggestion
   - `SERVICE_NAME`: Name of your Aiven for Apache Kafka service
```

File: docs/products/kafka/howto/kafka-sasl-auth.md [hunk 13]
```diff
@@ -1,28 +1,185 @@
...
+Configure SASL mechanisms for your Aiven for Apache Kafka service using
+[Aiven CLI](/docs/tools/cli):
+
+1. Get the name of the Aiven for Apache Kafka service:
+
+   ```bash
+   avn service list
+   ```
+
+  Note of the `SERVICE_NAME` corresponding to your Aiven for Apache Kafka service.
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/kafka-sasl-auth.md:121]:
```suggestion
  Note the `SERVICE_NAME` corresponding to your Aiven for Apache Kafka service.
```

File: docs/products/kafka/howto/kafka-sasl-auth.md [hunk 14]
```diff
@@ -1,28 +1,185 @@
...
+   ```bash
+   avn service update SERVICE_NAME             \
+    -c kafka_sasl_mechanisms.plain=true          \
+    -c kafka_sasl_mechanisms.scram_sha_256=true  \
+    -c kafka_sasl_mechanisms.scram_sha_512=true
+   ```
+
+   Parameters:
+
+   - `SERVICE_NAME`: Name of your Aiven for Apache Kafka service.
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/kafka-sasl-auth.md:134]:
```suggestion
   - `SERVICE_NAME`: Name of your Aiven for Apache Kafka service
```

File: docs/products/kafka/howto/kafka-sasl-auth.md [hunk 15]
```diff
@@ -1,28 +1,185 @@
...
+             }
+           }
+         }'
+```
+
+Parameters:
+
+- `project_name`: Name of your Aiven project.
+- `service_name`: Name of your Aiven for Apache Kafka service.
+- `API_TOKEN`: API token for authentication.
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/kafka-sasl-auth.md:164]:
```suggestion
- `project_name`: Name of your Aiven project
- `service_name`: Name of your Aiven for Apache Kafka service
- `API_TOKEN`: API token for authentication
```

File: docs/products/kafka/karapace/howto/enable-oauth-oidc-kafka-rest-proxy.md [hunk 16]
```diff
@@ -62,34 +68,46 @@ complete the following prerequisites and configuration steps:
...
 
-1. In [Aiven Console](https://console.aiven.io/), select your project,
-   and choose your Aiven for Apache Kafka® service.
-1. In the service page, select **Service settings** from the sidebar.
-1. On the **Service settings** page, scroll down to the **Advanced configuration**
-   section, and click **Configure**.
-1. In the **Advanced configuration** dialog, select **Add configuration options**.
-1. Look for  `kafka_rest_authorization` parameter and set it to `True`.
+1. In the [Aiven Console](https://console.aiven.io/), select your
+   project and choose your Aiven for Apache Kafka® service.
```
Comment (wojcik-dorota) [inline docs/products/kafka/karapace/howto/enable-oauth-oidc-kafka-rest-proxy.md:77]:
```suggestion
1. In the [Aiven Console](https://console.aiven.io/), select your
   project and Aiven for Apache Kafka® service.
```

File: docs/products/kafka/karapace/howto/enable-oauth-oidc-kafka-rest-proxy.md [hunk 17]
```diff
@@ -62,34 +68,46 @@ complete the following prerequisites and configuration steps:
...
+1. Click <ConsoleLabel name="Service settings"/>.
+1. Scroll to **Advanced configuration** and click **Configure**.
```
Comment (wojcik-dorota) [inline docs/products/kafka/karapace/howto/enable-oauth-oidc-kafka-rest-proxy.md:79]: nit:
```suggestion
1. Go to **Advanced configuration** and click **Configure**.
```

File: docs/products/kafka/howto/enable-oidc.md [hunk 18]
```diff
@@ -18,108 +31,111 @@ official documentation for specific configuration guidelines.
...
+- `kafka.sasl_oauthbearer_jwks_endpoint_url`
+  - **Description**: Endpoint for retrieving the JSON Web Key Set
+    (JWKS), which enables OIDC authentication. Corresponds to
+    the Apache Kafka parameter
+    `sasl.oauthbearer.jwks.endpoint.url`.
+  - **Value**: Enter the JWKS endpoint URL provided by your OIDC
+    provider.
+  - `kafka.sasl_oauthbearer_sub_claim_name` (Optional)
+    - **Description**: Name of the JWT's subject claim for broker
+      verification. This is optional and typically set to `sub`.
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/enable-oidc.md:105]: nit: you've already informed it's optional
```suggestion
      verification. It's typically set to `sub`.
```

File: docs/products/kafka/howto/enable-oidc.md [hunk 19]
```diff
@@ -18,108 +31,111 @@ official documentation for specific configuration guidelines.
...
+      Corresponds to the Apache Kafka parameter
+      `sasl.oauthbearer.sub.claim.name`.
+    - **Value**: Enter `sub` or the specific claim name provided
+      by your OIDC provider if different.
+  - `kafka.sasl_oauthbearer_expected_issuer` (Optional)
+    - **Description**: Specifies the JWT's issuer for the broker to
+      verify. Corresponds to the Apache Kafka parameter
+      `sasl.oauthbearer.expected.issuer`. This setting is
+      optional.
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/enable-oidc.md:114]:
```suggestion
      `sasl.oauthbearer.expected.issuer`.
```

File: docs/products/kafka/howto/enable-oidc.md [hunk 20]
```diff
@@ -18,108 +31,111 @@ official documentation for specific configuration guidelines.
...
+    - **Value**: Enter the issuer URL or identifier provided by your
+      OIDC provider.
+  - `kafka.sasl_oauthbearer_expected_audience` (Optional)
+    - **Description**: Validates the intended JWT audience for the
+      broker. Corresponds to the Apache Kafka parameter
+      `sasl.oauthbearer.expected.audience`. This is optional and
+      is used if your OIDC provider specifies an audience.
```
Comment (wojcik-dorota) [inline docs/products/kafka/howto/enable-oidc.md:121]:
```suggestion
      `sasl.oauthbearer.expected.audience`. It's used if your OIDC provider specifies an audience.
```

**Review by wojcik-dorota — COMMENTED** [review]
A few small fixes and some other very minor stuff :-)

**Comment by harshini-rangaswamy** [comment]
> A few small fixes and some other very minor stuff :-)

Thanks, @wojcik-dorota; I addressed most of them.

File: docs/products/kafka/howto/enable-oidc.md (hunk 4 above)
Comment (wojcik-dorota) [reply docs/products/kafka/howto/enable-oidc.md]: There still seems to be  a typo here: `Enabled`.

**Review by wojcik-dorota — APPROVED** [review]
LGTM! thanks :-)

**Pull request merged**